controller/watch/watcher.go

```diff
@@ -249,48 +249,70 @@ func (w *Watcher) Watch(t resourceType, ns string, wcb watchCallback) error {
 }
 
 func deploymentWatcher(di deployments.DeploymentInterface, wcb watchCallback) {
-	for {
-		log.Println("List all existing Deployments")
-		// First do List on the resource to bring things up to date.
-		l, err := di.List(api.ListOptions{})
-		for _, d := range l.Items {
-			log.Printf("Found Deployment name: %s\n", d.Name)
-			event := watch.Event{
-				Type:   watch.Added,
-				Object: &d,
+	// 'listChan' is our signal to know if we should re-get the full list
+	// of resources.
+	listChan := make(chan bool, 1)
```

Comment:
It's relative. It's not _too_ bad or _too_ costly, but it seems par for the course to obsess over efficiency in Go. I believe the way you can accomplish this without any unnecessary allocations is as follows:

```
listChan := make(chan interface{}, 1)
```